lib/iris/util.py

```diff
@@ -1554,3 +1555,100 @@ def _meshgrid(*xi, **kwargs):
         if mxii.dtype != xii.dtype:
             mxi[i] = mxii.astype(xii.dtype)
     return mxi
+
+
+def find_discontiguities(coord, abs_tol=1e-4):
+    """
+    Searches coord for discontiguities in the bounds array and masks the data
+    at those points if specified in kwargs.
+
+    Args:
+        coord:
+        Either
+
+        (a) An instance of :class:`iris.coords.Coord`
+
+        or
+
+        (b) the :attr:`standard_name`, :attr:`long_name`, or
+        :attr:`var_name` of an instance of an instance of
+        :class:`iris.coords.Coord`.
```

Comment:
If you are going to allow for a user to do this, you need to also add a line at the start that gets the coordinate.
This function starts by doing 
```
_, diffs_x, diffs_y = iris.coords._discontiguity_in_2d_bounds(coord.bounds,
                                                                  abs_tol)
```
so if that needs to be a coord instance by then

Reply:
Thing is, if we do allow that user to use a name instead of a coord object, then we have to pass the cube in as well so that we can get the coord.  This seems a bit unnecessary, so it might make sense to only allow a coord object.